Sources/Basics/FileSystem+Extensions.swift

```diff
@@ -224,26 +224,65 @@ extension FileSystem {
     }
 }
 
-// MARK: - cross-compilation SDKs
+// MARK: - cross-compilation destinations
 
 private let crossCompilationDestinationsDirectoryName = "destinations"
 
 extension FileSystem {
     /// SwiftPM cross-compilation destinations directory (if exists)
     public var swiftPMCrossCompilationDestinationsDirectory: AbsolutePath {
         get throws {
-            if let path = try self.idiomaticSwiftPMDirectory {
+            if let path = try idiomaticSwiftPMDirectory {
                 return path.appending(component: crossCompilationDestinationsDirectoryName)
             } else {
-                return try self.dotSwiftPMCrossCompilationDestinationsDirectory
+                return try dotSwiftPMCrossCompilationDestinationsDirectory
             }
         }
     }
 
     fileprivate var dotSwiftPMCrossCompilationDestinationsDirectory: AbsolutePath {
         get throws {
-            return try self.dotSwiftPM.appending(component: crossCompilationDestinationsDirectoryName)
+            return try dotSwiftPM.appending(component: crossCompilationDestinationsDirectoryName)
```

Comment:
Not sure this feels right. I think on macOS, we probably want this to be `~/Library/Application\ Support/org.swift.swiftpm` and not sure what a canonical place for other OSes would be.

Reply:
I'm modeling this after all other path options boilerplate, for which a symlink is created to `idiomaticSwiftPMDirectory` where possible.

Reply:
Sure, but isn't the primary location `~/Library/org.swift.swiftpm` right now?

Reply:
That's correct. Is your point that `dotSwiftPMCrossCompilationDestinationsDirectory` property should be deleted altogether? It's only called from `swiftPMCrossCompilationDestinationsDirectory`, which checks if `~/Library/org.swift.swiftpm` is a valid path and prefers that.

Reply:
I was wondering whether these should be in "Application Support" instead since they're more "data" than "configuration", but now looking at it, we should probably not have a `~/Library/org.swift.swiftpm` in the first place, but rather place things into subdirectories of "Preferences" and "Application Support"? Let's not worry about it for this PR.